its/tests/src/test/java/its/FileExclusionTests.java

```diff
@@ -0,0 +1,221 @@
+            Collections.emptyList(),
+            sonarUserHome.toString(),
+            Map.of(), false, null))
+        .get();
+    } catch (Exception e) {
+      throw new IllegalStateException("Cannot initialize the backend", e);
+    }
+  }
+
+  @BeforeAll
+  static void createSonarLintUser() {
+    adminWsClient = newAdminWsClient(ORCHESTRATOR);
+    adminWsClient.users().create(new CreateRequest().setLogin(SONARLINT_USER).setPassword(SONARLINT_PWD).setName("SonarLint"));
+  }
+
+  @BeforeEach
+  void clearState() {
+    didSynchronizeConfigurationScopes.clear();
+  }
+
+  @AfterEach
+  void removeConfigScope() {
+    backend.getConfigurationService().didRemoveConfigurationScope(new DidRemoveConfigurationScopeParams(CONFIG_SCOPE_ID));
+  }
+
+    @AfterAll
+  static void stop() throws ExecutionException, InterruptedException {
+    serverLauncher.getJavaImpl().shutdown().get();
+    System.clearProperty("sonarlint.internal.synchronization.initialDelay");
+    System.clearProperty("sonarlint.internal.synchronization.period");
+    System.clearProperty("sonarlint.internal.synchronization.scope.period");
+  }
+
+  @Test
+  void should_respect_exclusion_settings_on_SQ() {
+    var projectKey = "sample-java";
+    var projectName = "my-sample-java";
+    provisionProject(ORCHESTRATOR, projectKey, projectName);
+
+    backend.getConfigurationService().didAddConfigurationScopes(new DidAddConfigurationScopesParams(
+      List.of(new ConfigurationScopeDto(CONFIG_SCOPE_ID, null, true, projectName, new BindingConfigurationDto(CONNECTION_ID, projectKey,
+        true)))));
+    await().atMost(20, SECONDS).untilAsserted(() -> assertThat(didSynchronizeConfigurationScopes).contains(CONFIG_SCOPE_ID));
+
+    var filePath = Path.of("src/main/java/foo/Foo.java");
+    var clientFileDto = new ClientFileDto(filePath.toUri(), filePath, CONFIG_SCOPE_ID, null, StandardCharsets.UTF_8.name(),
+      filePath.toAbsolutePath(), null);
+    var didUpdateFileSystemParams = new DidUpdateFileSystemParams(List.of(), List.of(clientFileDto));
+    backend.getFileService().didUpdateFileSystem(didUpdateFileSystemParams);
+
+    // Firstly check file is included
+    var getFilesStatusParams = new GetFilesStatusParams(Map.of(CONFIG_SCOPE_ID, List.of(filePath.toUri())));
+    await().atMost(10, SECONDS).untilAsserted(() ->
+      assertThat(backend.getFileService().getFilesStatus(getFilesStatusParams).get().getFileStatuses().get(filePath.toUri()).isExcluded()).isFalse());
+
+    // Change file exclusion settings on SQ which should not affect Foo.java
+    adminWsClient.settings().set(new SetRequest()
+      .setKey("sonar.exclusions")
+      .setValues(singletonList("**/*.js"))
+      .setComponent(projectKey));
+
+    // Check Foo.java is still included
+    await().atMost(10, SECONDS).untilAsserted(() ->
+      assertThat(backend.getFileService().getFilesStatus(getFilesStatusParams).get().getFileStatuses().get(filePath.toUri()).isExcluded()).isFalse());
```

Comment:
I think this assertion is useless because anyway, it is unlikely the new exclusion would be processed that fast.

What you could do is move this case after a case where the file was excluded:

assert excluded = false

set sonar.exclusion = **/*.java
await excluded = true

set sonar.exclusion = **/*.js
await excluded = false

set sonar.inclusion = **/*.js
await excluded = true

reset sonar.inclusion and sonar.exclusion
await excluded = false